core/message_handler_test.go

```diff
@@ -0,0 +1,230 @@
+	defer meshDB.Close()
+	require.NoError(t, err)
+
+	selector := &OrderSelector{
+		nextOffset: 0,
+		db:         meshDB,
+	}
+
+	// Add five orders to the database
+	orders := randomOrders(t, 5)
+	allOrders = orders
+	insertOrders(t, selector, orders)
+	verifyRoundRobinSharing(t, selector, 0, 3)
+
+	// Add seven more orders to the database
+	orders = randomOrders(t, 7)
+	allOrders = append(allOrders, orders...)
+	insertOrders(t, selector, orders)
+	verifyRoundRobinSharing(t, selector, 3, 10)
+
+	// Delete three orders from the database
+	deleteOrders(t, selector, allOrders[0:3:3])
+	verifyRoundRobinSharing(t, selector, selector.nextOffset, 5)
+
+	// Add 12 more orders to the database
+	orders = randomOrders(t, 12)
+	allOrders = append(allOrders, orders...)
+	insertOrders(t, selector, orders)
+	verifyRoundRobinSharing(t, selector, selector.nextOffset, 7)
+}
+
+// Verify that the correct messages are shared by `GetMessagesToShare` given `orders`, a `nextOffset`, and `max`
+func verifyRoundRobinSharing(t *testing.T, selector *OrderSelector, nextOffset int, max int) {
+	notRemovedFilter := selector.db.Orders.IsRemovedIndex.ValueFilter([]byte{0})
+	ordersSnapshot, err := selector.db.Orders.GetSnapshot()
+
+	// Get the number of orders in the database
+	count, err := ordersSnapshot.NewQuery(notRemovedFilter).Count()
+	require.NoError(t, err)
+
+	expectedOrdersLength := min(max, count)
+	expectedOrders := make([][]byte, expectedOrdersLength)
+
+	// Get all of the orders in the database.
+	var orderList []*meshdb.Order
+	err = ordersSnapshot.NewQuery(notRemovedFilter).Offset(0).Max(count).Run(&orderList)
+
+	// Update `nextOffset` to zero if it is larger than the number of orders that are stored
+	if nextOffset > count {
+		nextOffset = 0
+	}
+
+	// Calculate the orders that we expect to be shared
+	for i := 0; i < expectedOrdersLength; i++ {
+		encoding, err := encodeOrder(orderList[(nextOffset+i)%count].SignedOrder)
+		require.NoError(t, err)
+
+		expectedOrders[i] = encoding
+	}
+
+	// Get the actual list of orders that are shared
+	actualOrders, err := selector.GetMessagesToShare(max)
+	require.NoError(t, err)
+
+	// Ensure that the result from `GetMessagesToShare` matches the expected result.
+	assert.Equal(t, expectedOrders, actualOrders)
+}
+
+func deleteOrders(t *testing.T, selector *OrderSelector, orders []*meshdb.Order) {
+	for _, order := range orders {
+		err := selector.db.Orders.Delete(order.ID())
+		require.NoError(t, err)
+	}
+}
+
+func insertOrders(t *testing.T, selector *OrderSelector, orders []*meshdb.Order) {
+	for _, order := range orders {
+		err := selector.db.Orders.Insert(order)
+		require.NoError(t, err)
+	}
+}
+
+func randomOrders(t *testing.T, orderCount int) []*meshdb.Order {
+	orders := make([]*meshdb.Order, orderCount)
+
+	for i := 0; i < orderCount; i++ {
+		orders[i] = randomOrder(t)
+	}
+
+	return orders
+}
+
+func randomOrder(t *testing.T) *meshdb.Order {
```

Comment:
Let's use the `scenario` package here instead of re-writing functions for generating orders. I don't really see any reason to randomize the order fields since the method we're testing doesn't look at the order fields or process them at all.